### PR TITLE
 Fix to stale or incorrect trashed collections data appearing in librariesCollectionsBox

### PR DESCRIPTION
**Pull request by abaevbog**
- listen to `collection` notifier events and re-render  the section if a relevant collection is trashed or modified. That way, a trashed collection will be removed, restored collection will be added back and renaming a collection will update the name in the librariesCollectionsBox.
  - fixed a bug where a restored collection would not be added into `item._collections` cache. Fixed false-positive test for it.
  - do not add trashed collections into `item. _collections` cache in `Zotero.Items._loadCollections`. Otherwise, trashed collections will appear in librariesCollectionsBox after the app is restarted.

  Fixes: #4307

### CODE REVIEW

File: chrome/content/zotero/elements/librariesCollectionsBox.js [hunk 1]
```diff
@@ -77,11 +77,19 @@ import { getCSSIcon } from 'components/icons';
...
+			if (!this._item) return;
 			if (action == 'modify'
-					&& this._item
+					&& type == "item"
 					&& (ids.includes(this._item.id) || this._linkedItems.some(item => ids.includes(item.id)))) {
 				this._forceRenderAll();
 			}
+			
+			if (["modify", "trash"].includes(action) && type == "collection") {
+				let isRelevantCollection = ids.findIndex(id => Zotero.Collections.get(id).hasItem(this._item)) !== -1;
```
Comment (dstillman) [inline chrome/content/zotero/elements/librariesCollectionsBox.js:88]: `ids.some()` would probably be clearer

**Review by dstillman — CHANGES_REQUESTED** [review]
We should have a test for the Libraries & Collections box too, not just the cache issue.

File: chrome/content/zotero/elements/librariesCollectionsBox.js (hunk 1 above)
Comment (abaevbog) [reply chrome/content/zotero/elements/librariesCollectionsBox.js]: Yes, certainly

**Comment by abaevbog** [comment]
I just pushed a few tests to cover these behaviors for `librariesCollectionsBox` as well. On `main`, all of these tests would currently fail as expected

**Pull request merged**